server/pairing/raw_message_handler.go

```diff
@@ -0,0 +1,175 @@
+package pairing
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"path/filepath"
+
+	"github.com/golang/protobuf/proto"
+	"github.com/google/uuid"
+
+	"github.com/status-im/status-go/api"
+	"github.com/status-im/status-go/multiaccounts"
+	"github.com/status-im/status-go/multiaccounts/accounts"
+	"github.com/status-im/status-go/params"
+	"github.com/status-im/status-go/protocol/protobuf"
+	"github.com/status-im/status-go/server"
+)
+
+const (
+	logLevel    = "ERROR"
+	dataDir     = "ethereum/mainnet_rpc"
+	upstreamURL = "https://mainnet.infura.io/v3/800c641949d64d768a5070a1b0511938"
+	fleet       = "eth.prod"
+	nodeName    = "StatusIM"
+)
+
+type SyncRawMessageHandler struct {
+	backend *api.GethStatusBackend
+}
+
+func NewSyncRawMessageHandler(backend *api.GethStatusBackend) *SyncRawMessageHandler {
+	return &SyncRawMessageHandler{backend: backend}
+}
+
+func (s *SyncRawMessageHandler) PrepareRawMessage(keyUID string) ([]byte, error) {
+	messenger := s.backend.Messenger()
+	if messenger == nil {
+		return nil, fmt.Errorf("messenger is nil when handlePairingSyncDeviceSend")
+	}
+
+	currentAccount, err := s.backend.GetActiveAccount()
+	if err != nil {
+		return nil, err
+	}
+	if keyUID != currentAccount.KeyUID {
+		return nil, fmt.Errorf("keyUID not equal")
+	}
+
+	rawMessageCollector := new(server.RawMessageCollector)
+	err = messenger.SyncDevices(context.TODO(), currentAccount.Name, currentAccount.Identicon, rawMessageCollector)
+	if err != nil {
+		return nil, err
+	}
+	syncRawMessage := new(protobuf.SyncRawMessage)
+	for _, m := range rawMessageCollector.GetRawMessages() {
+		rawMessage := new(protobuf.RawMessage)
+		rawMessage.Payload = m.Payload
+		rawMessage.MessageType = m.MessageType
+		syncRawMessage.RawMessages = append(syncRawMessage.RawMessages, rawMessage)
+	}
+
+	subAccounts, err := s.backend.StatusNode().AccountService().GetAccountsByKeyUID(keyUID)
+	if err != nil {
+		return nil, err
+	}
+	subAccountsJSONBytes, err := json.Marshal(subAccounts)
+	if err != nil {
+		return nil, err
+	}
+	syncRawMessage.SubAccountsJsonBytes = subAccountsJSONBytes
+	return proto.Marshal(syncRawMessage)
+}
+
+func (s *SyncRawMessageHandler) HandleRawMessage(account *multiaccounts.Account, password, keystorePath string, payload []byte) error {
+	var syncRawMessage protobuf.SyncRawMessage
+	err := proto.Unmarshal(payload, &syncRawMessage)
+	if err != nil {
+		return err
+	}
+
+	newKeystoreDir := filepath.Join(keystorePath, account.KeyUID)
+	accountManager := s.backend.AccountManager()
+	err = accountManager.InitKeystore(newKeystoreDir)
+	if err != nil {
+		return err
+	}
+
+	err = s.saveSubAccounts(account, password, &syncRawMessage)
+	if err != nil {
+		return err
+	}
+
+	nodeConfig, err := s.defaultNodeConfig(keystorePath, account.KeyUID)
+	if err != nil {
+		return err
+	}
+	err = s.backend.StartNodeWithAccount(*account, password, nodeConfig)
```

Comment:
oh, i should call StartNodeWithAccountAndInitialConfig instead, and i should get settings from sender ?